src/main/kotlin/nomic/domain/games/GameDomain.kt

```diff
@@ -25,6 +30,22 @@ class GameDomain(private val gameRepository: GameRepository) : IGameDomain {
         gameRepository.createGame(input)
     }
 
+    override fun listGames(user: EndUser, size: UInt, offset: UInt): Optional<List<GameModel>> {
+        // Max size and offset conditions are endpoint logic rather than business logic.
+
+        val games = gameRepository.listGames(
+            specFactory.filterByUser(user),
+            specFactory.sortByMostRecentFirst(),
+            specFactory.paginate(size, offset)
+        )
+
+        if (offset > 0u && games.isEmpty()) {
+            return Optional.empty()
+        }
```

Comment:
Is there a reason that option is needed here?  The system could just as easily throw an exception for the globalExceptionHandler to manage.  Rather than offloading that work onto the endpoint